source/agora/utils/TxBuilder.d

```diff
@@ -319,20 +319,22 @@ public struct TxBuilder
         foreach (ref o; this.data.outputs)
             o.type = outputs_type;
 
+        // Sign all inputs using unlocker
+        foreach (idx, ref in_; this.inputs)
+            this.data.inputs[idx].unlock = unlocker(this.data, in_);
+
         // Add the refund output if needed
-        if (this.leftover.value > Amount(0))
+        if (this.leftover.value > fees)
```

Comment:
`fee` should not be an absolute value but a ration per byte, and this function needs to compute the correct size.

Reply:
Yes as a first step I thought we could set fees high enough to prevent all txs being rejected. Then introduce a calculated fee based on the size of the tx.

Reply:
I think going straight for the calculated fee is better. A set fee is just an extra output with some specific value, we can already do that manually. Plus, we won't have to change the tests twice.

Reply:
There is no output for the fee. Just the leftover from total input values - total output values. I almost thinking we don't need to calculate in TxBuilder as it is just generating test transactions. The calculated fee per byte happens when we validate and decide on candidates.